### PR TITLE
[rpm/deb] Initial work for creating RPM and Debian packages

### PR DESCRIPTION
**Pull request by bhavin192**
- This work is based on #11
- Keeping name of the packages as `yugabytedb` and `yugabytedb-client`.
- The data is stored in `/var/lib/yugabytedb/` and logs are stored in `/var/log/yugabytedb`
- Creates the `yugabyte` user on the system
- Adds scripts to test the package installation (checks if symlinks are correct, systemd service is in correct state and database creation and retrieval works using ysqlsh)

### CODE REVIEW

File: packages/yugabyte-client-package-script/install.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+if [ -d "/opt/yugabyte/" ]; then
+  if [ -L "/usr/bin/cqlsh" ] && [ -L "/usr/bin/ysqlsh" ]; then
+    rm -f /usr/bin/ysqlsh
+    rm -f /usr/bin/cqlsh
+    cd /opt/yugabyte-client/ || exit 1
+    ./bin/post_install.sh
+    ln -s /opt/yugabyte-client/bin/cqlsh /usr/bin/cqlsh
+    ln -s /opt/yugabyte-client/postgres/bin/ysqlsh /usr/bin/ysqlsh
+  else
+    cd /opt/yugabyte-client/ || exit 1
```
Comment (iSignal) [inline packages/yugabyte-client-package-script/install.sh]: minor: move the common parts of the if and else out of the else

File: packages/yugabyte-client-package-script/post_upgrade.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+yugabytedb_client_path="/opt/yugabytedb-client"
+
+set -x
+# post_install.sh is required after upgrade of the package
+if [ -f "${yugabytedb_client_path}/.post_install.sh.completed" ]; then
+  rm "${yugabytedb_client_path}/.post_install.sh.completed"
```
Comment (iSignal) [inline packages/yugabyte-client-package-script/post_upgrade.sh]: great catch!

File: packages/yugabyte-client-package-script/install.sh (hunk 1 above)
Comment (bhavin192) [reply packages/yugabyte-client-package-script/install.sh]: Yep, you are correct. I'm going to remove this file actually as we have post_install.sh.

File: packages/server/yugabyted.service [hunk 3]
```diff
@@ -0,0 +1,15 @@
+[Unit]
+Description=Daemon to start a single-node YugabyteDB process
+Documentation=https://docs.yugabyte.com
+
+[Service]
+Type=forking
+User=yugabyte
+ExecStart=/usr/bin/python /opt/yugabytedb/bin/yugabyted start --config /etc/yugabytedb/yugabytedb.conf
+PIDFile=/opt/yugabytedb/yugabyted.pid
```
Comment (iSignal) [inline packages/server/yugabyted.service:9]: I think this pid file location is going to move to the data directory /var/lib/yugabytedb/yugabyte-data/ in the next release. we don't need to update it right now but just an fyi.

File: packages/server/yugabytedb.conf [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{
+  "tserver_webserver_port": 9000,
+  "master_rpc_port": 7100,
+  "webserver_port": 7200,
+  "master_webserver_port": 7000,
+  "bind_ip": "127.0.0.1",
```
Comment (iSignal) [inline packages/server/yugabytedb.conf:6]: bind_ip should default to 0.0.0.0

**Review by iSignal — APPROVED** [review]

**Pull request merged**